scripts/build/uf2conv.py

```diff
@@ -242,36 +255,57 @@ def write_file(name, buf):
     print("Wrote %d bytes to %s" % (len(buf), name))
 
 
+def load_families():
+    # The expectation is that the `uf2families.json` file is in the same
+    # directory as this script. Make a path that works using `__file__`
+    # which contains the full path to this script.
+    filename = "uf2families.json"
+    pathname = os.path.join(os.path.dirname(os.path.abspath(__file__)), filename)
+    with open(pathname) as f:
+        raw_families = json.load(f)
+
+    families = {}
+    for family in raw_families:
+        families[family["short_name"]] = int(family["id"], 0)
+
+    return families
+
+
 def main():
     global appstartaddr, familyid
     def error(msg):
-        print(msg)
+        print(msg, file=sys.stderr)
         sys.exit(1)
-    parser = argparse.ArgumentParser(description='Convert to UF2 or flash directly.',
-                                     allow_abbrev=False)
+    parser = argparse.ArgumentParser(description='Convert to UF2 or flash directly.')
```

Comment:
this should have `allow_abbrev=False` there is something wrong with the CI because this should have failed CI

Reply:
as this is a tool coming from a pure external repo and copied verbatim into Zephyr, then I don't think we should be letting it fail our CI because of lint warnings.

I see no principle difference between the current approach and then installing a package through pip.
(where we have no linting or control over arg abbrev)

In fact, if uf2conv had been available as a pip package we would probably have had it in requirements.txt instead.